src/ComparedQueryable/NativeQueryable/EnumerableRewriter.cs

```diff
@@ -54,13 +56,13 @@ protected override Expression VisitMethodCall(MethodCallExpression m)
             return m;
         }
 
-        private ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
+        protected virtual ReadOnlyCollection<Expression> FixupQuotedArgs(MethodInfo mi, ReadOnlyCollection<Expression> argList)
         {
             ParameterInfo[] pis = mi.GetParameters();
             if (pis.Length > 0)
             {
                 List<Expression> newArgs = null;
-                for (int i = 0, n = pis.Length; i < n; i++)
+                for (int i = 0, n = argList.Count; i < n; i++)

```

Comment:
I think here pis.Length should be equal to argList.Count? Why make the change in line 63/65? If the lengths are the same, it really doesn't matter, but you're checking one length and then looping based on another, which is unexpected and doesn't seem to improve the code.